src/Updates.elm

```diff
@@ -0,0 +1,148 @@
+module Updates exposing (update)
+
+import Alerts.Api
+import Navigation
+import Silences.Types exposing (nullSilence)
+import Task
+import Types exposing (Msg(..), Model, Route(NotFoundRoute, SilenceFormEditRoute, SilenceFormNewRoute, SilenceRoute, StatusRoute, SilenceListRoute, AlertsRoute))
+import Utils.List
+import Utils.Types
+    exposing
+        ( ApiResponse(Loading, Failure, Success)
+        , Matcher
+        , nullFilter
+        )
+import Views.AlertList.Updates
+import Views.Silence.Types exposing (SilenceMsg(SilenceFetched, InitSilenceView))
+import Views.Silence.Updates
+import Views.SilenceForm.Types exposing (SilenceFormMsg(NewSilence, FetchSilence))
+import Views.SilenceForm.Updates
+import Views.SilenceList.Updates
+import Views.Status.Types exposing (StatusMsg(InitStatusView))
+import Views.Status.Updates
+import String exposing (trim)
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        CreateSilenceFromAlert alert ->
+            let
+                silence =
+                    { nullSilence | matchers = (List.map (\( k, v ) -> Matcher k v False) alert.labels) }
+            in
+                ( { model | silence = Success silence }, Cmd.none )
+
+        PreviewSilence silence ->
+            let
+                s =
+                    { silence | silencedAlertGroups = Loading }
+
+                filter =
+                    { nullFilter | text = Just <| Utils.List.mjoin s.matchers }
+            in
+                ( { model | silence = Success silence }, Alerts.Api.alertPreview filter )
+
+        AlertGroupsPreview alertGroups ->
+            let
+                silence =
+                    case model.silence of
+                        Success sil ->
+                            Success { sil | silencedAlertGroups = alertGroups }
+
+                        Failure e ->
+                            Failure e
+
+                        Loading ->
+                            Loading
+            in
+                ( { model | silence = silence }, Cmd.none )
+
+        NavigateToAlerts alertsRoute ->
+            let
+                ( alertsMsg, filter ) =
+                    (Views.AlertList.Updates.urlUpdate alertsRoute)
```

Comment:
Note to self: This can probably be removed since there's only a single route, the `AlertsGroup` route, being used. When we add `id` to the alerts and have an individual detail page for them, we'll follow the pattern established in this PR and give them their own route and view model.